stan/math/prim/meta/is_vector.hpp

```diff
@@ -44,6 +45,37 @@ struct is_eigen_row_vector_impl
  */
 template <typename T>
 struct is_eigen_row_vector_impl<T, false> : std::false_type {};
+
+/** \ingroup type_trait
+ * Underlying implementation for detecting if a Matrix is a column vector.
+ */
+template <typename T, bool = bool_constant<is_eigen<T>::value
```

Comment:
I like is_matrix_like but for these since they just check that Rows/ColsAtComplileTime are equal to 1 I kind of like leaving these as is_col_vector etc. We have a is_vector_like already (which just checks that the type has a operator[](int) so it works for pointers as well). Maybe we should change that definition too?